src/lib/storage/column_iterables/chunk_offset_mapping.cpp

```diff
@@ -18,4 +21,34 @@ ChunkOffsetsByChunkID split_pos_list_by_chunk_id(const PosList& pos_list) {
   return chunk_offsets_by_chunk_id;
 }
 
+std::optional<std::pair<ChunkID, ChunkOffsetsList>> to_chunk_id_and_chunk_offsets_list(const PosList& pos_list) {
+  auto chunk_offsets_list = ChunkOffsetsList{};
+  chunk_offsets_list.reserve(pos_list.size());
+
+  // Find first row ID that is not NULL
+  const auto first_not_null_it = std::find_if(pos_list.cbegin(), pos_list.cend(),
+                                              [](const auto& row_id) { return !row_id.is_null(); });
+
+  // If none was found, return empty list
+  if (first_not_null_it == pos_list.cend()) return std::nullopt;
+
+  const auto chunk_id = first_not_null_it->chunk_id;
+
+  // Skip any row ID that is NULL.
+  const auto first_chunk_offset = static_cast<ChunkOffset>(std::distance(pos_list.cbegin(), first_not_null_it));
+  for (auto chunk_offset = first_chunk_offset; chunk_offset < pos_list.size(); ++chunk_offset) {
+    const auto row_id = pos_list[chunk_offset];
+
+    // Skip any row ID that is NULL.
+    if (row_id.is_null()) continue;
+
+    // Fail if PosList references more than one chunk
+    if (row_id.chunk_id != chunk_id) return std::nullopt;
+
+    chunk_offsets_list.push_back({chunk_offset, row_id.chunk_offset});
```

Comment:
Maybe. Since the two values are integers, I guess the additional copies are removed by the compiler anyway. Unless we can verify the performance improvement, I wouldn’t change it.

Reply:
You're favouring last millennium's`push_back` over explicitly specifying you intend of constructing an object in place by using `emplace_back`? Wow, wouldn't have expected that.

Reply:
`emplace_back` has also disadvantages. It ignores constructors declared `explicit`. I think `emplace_back` is meant for those cases when the objects in the vector have complex (copy) constructors, so in order to avoid additional overhead, you can use `emplace_back`. For such primitive PODs as well as pointer types, I don’t think there is any benefit. But you can always prove me wrong!